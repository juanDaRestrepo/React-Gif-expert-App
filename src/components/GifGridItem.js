import React from 'react'

export const GifGridItem = ({id,url,title}) => {
    
    return (
        <div className='card animate__animated animate__backInLeft '>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
