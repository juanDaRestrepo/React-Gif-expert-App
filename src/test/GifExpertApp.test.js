import { shallow } from "enzyme";
import { GifGrid } from "../components/GifGrid";
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas en en el componente GifExpertApp' , () => {
    test('debe mostrarse el componente correctamente el componente', () => { 
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostar un determinado numero de categorias por defecto', () => {
        
        const categories = ['One Punch','Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);        
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
})