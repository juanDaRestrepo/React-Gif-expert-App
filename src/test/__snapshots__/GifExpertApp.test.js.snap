// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en en el componente GifExpertApp debe mostar un determinado numero de categorias por defecto 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon ball"
    key="Dragon ball"
  />
</div>
`;

exports[`Pruebas en en el componente GifExpertApp debe mostrarse el componente correctamente el componente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</div>
`;
